docker/build-tools/Dockerfile

```diff
@@ -553,27 +614,69 @@ RUN apt-get update && apt-get install -y --no-install-recommends \
  && apt-get clean \
  && rm -rf /var/lib/apt/lists/*
 
+# 11.0.1 is the version support ubuntu:xenial & aarch64
 ENV LLVM_VERSION=11.0.1
-ENV LLVM_TARGET=x86_64-linux-gnu-ubuntu-16.04
 ENV LLVM_BASE_URL=https://github.com/llvm/llvm-project/releases/download/llvmorg-${LLVM_VERSION}
-ENV LLVM_ARCHIVE=clang+llvm-${LLVM_VERSION}-${LLVM_TARGET}
-ENV LLVM_URL=${LLVM_BASE_URL}/${LLVM_ARCHIVE}.tar.xz
 ENV LLVM_DIRECTORY=/usr/lib/llvm
 
-RUN wget ${LLVM_URL}
-RUN tar -xJf ${LLVM_ARCHIVE}.tar.xz -C /tmp
-RUN mkdir -p ${LLVM_DIRECTORY}
-RUN mv /tmp/${LLVM_ARCHIVE}/* ${LLVM_DIRECTORY}/
+RUN set -eux; \
+    \
+    case $(uname -m) in \
+        x86_64) export LLVM_ARCHIVE=clang+llvm-${LLVM_VERSION}-x86_64-linux-gnu-ubuntu-16.04;; \
+        aarch64) export LLVM_ARCHIVE=clang+llvm-${LLVM_VERSION}-aarch64-linux-gnu;; \
+        *) echo "unsupported architecture"; exit 1 ;; \
+    esac; \
+    \
+    wget ${LLVM_BASE_URL}/${LLVM_ARCHIVE}.tar.xz; \
+    tar -xJf ${LLVM_ARCHIVE}.tar.xz -C /tmp; \
+    mkdir -p ${LLVM_DIRECTORY}; \
+    mv /tmp/${LLVM_ARCHIVE}/* ${LLVM_DIRECTORY}/
+
+###########
+# GN
+###########
+
+FROM debian:buster AS gn_context
+
+RUN set -eux; \
+    \
+    apt-get update; \
+    apt-get install -qqy --no-install-recommends \
+        ca-certificates git \
+        clang python ninja-build \
+        libclang-dev libc++-dev \
+        ; \
+    rm -rf /var/lib/apt/lists/*
+
+WORKDIR /tmp
+RUN git clone https://gn.googlesource.com/gn;
+
+WORKDIR /tmp/gn
+
+RUN set -eux; \
+    \
+    git checkout 501b49a3; \
+    python build/gen.py; \
+    ninja -v -C out; \
+    out/gn_unittests; \
+    mkdir -p /gn; \
+    cp /tmp/gn/out/gn /gn/gn; \
+    /gn/gn --version;
 
 ###########
 # Bazel
 ###########
 
-FROM ubuntu:xenial as bazel_context
+FROM ubuntu:xenial AS bazel_context_amd64
+FROM ubuntu:focal AS bazel_context_arm64
+# hadolint ignore=DL3006
+FROM bazel_context_${TARGETARCH} AS bazel_context
 
-ENV BAZELISK_VERSION="v1.0"
+ARG TARGETARCH
+
+ENV BAZELISK_VERSION="v1.9.0"
 ENV BAZELISK_BASE_URL="https://github.com/bazelbuild/bazelisk/releases/download"
-ENV BAZELISK_BIN="bazelisk-linux-amd64"
+ENV BAZELISK_BIN="bazelisk-linux-${TARGETARCH}"
```

Comment:
Is this TARGETARCH or arch of host?

Reply:
`BUILDARCH` is arch of host, `TARGETARCH` if from `--platform`
https://docs.docker.com/engine/reference/builder/#automatic-platform-args-in-the-global-scope